### PR TITLE
crd: introduce disk driver in node.spec.diskSpec

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7672

### CODE REVIEW

File: chart/templates/crds.yaml [hunk 1]
```diff
@@ -2116,6 +2116,12 @@ spec:
                       - filesystem
                       - block
                       type: string
+                    diskDriver:
```
Comment (innobead) [inline chart/templates/crds.yaml:2119]: Can we make a default value here?

Comment (derekbit) [reply chart/templates/crds.yaml:2119]: No, it is not allowable.

Comment (innobead) [reply chart/templates/crds.yaml:2119]: I see, so we will assume still `aio` will be the default, correct?

Comment (derekbit) [reply chart/templates/crds.yaml:2119]: We will use `auto` in my current implementation.
If the diskPath is a BDF, the SPDK engine discerns whether it's an NVMe or Virtio disk. Alternatively, if the path follows a canonical format such as `/dev/sdb`, it indicates an `aio` disk.

Comment (innobead) [reply chart/templates/crds.yaml:2119]: Sounds good :muscle: , but we should show the real drive info in the status.

Comment (derekbit) [reply chart/templates/crds.yaml:2119]: Yes, the information will be recorded in the status
image

**Review by innobead — APPROVED** [review]

**Pull request merged**